sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/datasource/notification/create/FetchSubscriberDbConnector.groovy

```diff
@@ -0,0 +1,103 @@
+package life.qbic.samplenotificator.datasource.notification.create
+
+import groovy.util.logging.Log4j2
+import life.qbic.business.exception.DatabaseQueryException
+import life.qbic.business.subscription.fetch.FetchSubscriberDataSource
+import life.qbic.business.subscription.Subscriber
+import life.qbic.datamodel.samples.Status
+import life.qbic.samplenotificator.datasource.database.ConnectionProvider
+import life.qbic.samplenotificator.datasource.database.DatabaseSession
+
+import java.sql.Connection
+import java.sql.PreparedStatement
+import java.sql.Timestamp
+import java.time.Instant
+import java.time.LocalDate
+import java.time.ZoneId
+
+/**
+ * <h1>A database connector to retrieve information for sending notifications</h1>
+ *
+ * <p>This connector reads the notification queue table and finds the associated people that subscribed for a notification</p>
+ *
+ * @since 1.0.0
+ *
+*/
+@Log4j2
+class FetchSubscriberDbConnector implements FetchSubscriberDataSource{
+    private ConnectionProvider connectionProvider
+
+    FetchSubscriberDbConnector(ConnectionProvider connectionProvider){
+        this.connectionProvider = connectionProvider
+    }
+
+    @Override
+    Map<String, Status> getNotificationsForDay(LocalDate day){
```

Comment:
You are right, I will update the names. I hope its ok for you if I rename it to getUpdatedSamples(LocalDate day). I think its fine to use day since the date reflects a day.